### DISCUSSION TITLE
can't find correct device declaration for NodeMCU and ILI9341 SPI....

### DISCUSSION THREAD

**Discussion by hugofant** [moononournation/Arduino_GFX#192] [General]
everythings works well with the original ADAFruit libaries

`#include "Adafruit_GFX.h"
#include "Adafruit_ILI9341.h"

#define TFT_DC 12
#define TFT_CS 15
#define TFT_RST 13
#define TFT_MISO 4
#define TFT_MOSI 14
#define TFT_CLK 2

Adafruit_ILI9341 tft = Adafruit_ILI9341(TFT_CS, TFT_DC, TFT_MOSI, TFT_CLK, TFT_RST, TFT_MISO);`

Cause of performance (and also a bit of curiosity) i wanted to try out this library....but i'm not able to find out the correct  device declaration.
some help/info would be great.

thanks in advance and best regards

**Comment by moononournation** [comment]
It may just fast a little bit. If you still want to fulfill your curiosity, try:
```
#include <Arduino_GFX_Library.h>
Arduino_DataBus *bus = new Arduino_HWSPI(TFT_DC, TFT_CS, TFT_CLK, TFT_MOSI, GFX_NOT_DEFINED /* MISO */);
Arduino_GFX *gfx = new Arduino_ILI9341(bus, TFT_RST);
```

**Reply by hugofant** [reply]
thanks for answering that quick.
I've also tried with something similar to this and it results in a compiler error:
`HelloWorld:37:103: error: no matching function for call to 'Arduino_HWSPI::Arduino_HWSPI(int, int, int, int, int)'
   37 | Arduino_DataBus *bus = new Arduino_HWSPI(TFT_DC, TFT_CS, TFT_CLK, TFT_MOSI, GFX_NOT_DEFINED /* MISO */);
      |                                                                                                       ^
`

...and shoudn't it be:
Arduino_DataBus *bus = new Arduino_HWSPI(TFT_DC, TFT_CS, TFT_CLK, TFT_MOSI, **TFT_MISO** /* MISO */);

**Reply by hugofant** [reply]
forgot ti mention: it's an ili9341 TFT SPI TFT with 240x320 resolution - Model Number: kmrtm28028-SPI

**Reply by moononournation** [reply]
if you are using AVR dev board, simply pass first 2 parameters:
Arduino_DataBus *bus = new Arduino_HWSPI(TFT_DC, TFT_CS);

**Comment by hugofant** [comment]
with this settings compiler error is gone....but display stays white without showing anything....
Whats the idea behind? How to set CLK, MOSI and MISO - or is the only way to use the default pins?

**Reply by moononournation** [reply]
can you tell me what dev board are you using?

**Reply by hugofant** [reply]
have already written (in headline) -  it's a NodeMCU (ESP8266).

**Reply by moononournation** [reply]
ESP8266 cannot remap the SPI pins

**Reply by moononournation** [reply]
and the example file header show the default pins for each type of dev board. Please try to use default pins first before change the pins.

**Comment by hugofant** [comment]
ok, just for my better understanding - so, it's not my "fault" or the ESP8266's, it's cause remapping is not possible when using your library in combination with ESP8266.
Correct?

**Reply by moononournation** [reply]
Just because ESP8266 does not have this features.

**Reply by hugofant** [reply]
??? hmm, i don't understand that - with the original "Adafruit_ILI9341.h" library it's working well.

**Reply by moononournation** [reply]
I think that is software SPI and much slower than hardware SPI. But you are concerning the speed, right?
